### PR TITLE
Support sqlite pragma statements

### PR DESCRIPTION
**Pull request by tami5**

### CODE REVIEW

File: lua/sql.lua [hunk 1]
```diff
@@ -16,12 +16,36 @@ function sql:__assert_tbl(tbl, method)
...
+  "foreign_key_list","foreign_keys","freelist_count","fullfsync","function_list",
+  "hard_heap_limit","ignore_check_constraints","incremental_vacuum","index_info","index_list",
+  "index_xinfo","integrity_check","journal_mode","journal_size_limit","legacy_alter_table",
+  "legacy_file_format","locking_mode","max_page_count","mmap_size","module_list","optimize",
+  "page_count","page_size","parser_trace","pragma_list","query_only","quick_check",
+  "read_uncommitted","recursive_triggers","reverse_unordered_selects","schema_version","secure_delete",
+  "shrink_memory","soft_heap_limit","stats","synchronous","table_info","table_xinfo","temp_store",
+  "threads","trusted_schema","user_version","vdbe_addoptrace²","vdbe_debug²","vdbe_listing",
+  "vdbe_trace","wal_autocheckpoint","wal_checkpoint","writable_schema"
+}
```
Comment (Conni2461) [inline lua/sql.lua:38]: Pro tip to do that in O(1) rather then O(n) or O(log(n)) when using binary search.
```lua
local valid_paragma = {
  ["analysis_limit"] = true,
  ["application_id"] = true,
  ["auto_vacuum"] = true
}

local valid_paragma_key = function(value)
  return valid_paragma[value] ~= nil
end
```

File: lua/sql.lua [hunk 2]
```diff
@@ -54,33 +83,35 @@ end
...
+      self:__connect()
+    end
     if not self.closed then
       return self
     else
       return false
     end
   end
 
+  print(vim.inspect(opts))
```
Comment (Conni2461) [inline lua/sql.lua:100]: Debug code :) Just putting that here so it will not be shipped

File: lua/sql.lua [hunk 3]
```diff
@@ -54,33 +83,35 @@ end
...
     o.closed = true
     return o
   end
   o.modified = false
+
   o:__connect()
   return o
 end
-
+print(vim.inspect(sql:open "/tmp/dbdbdb.sql", {jfdfd = '1'}))
```
Comment (Conni2461) [inline lua/sql.lua:114]: Debug code

File: lua/sql.lua (hunk 1 above)
Comment (tami5) [reply lua/sql.lua:38]: :heart_eyes: :heart_eyes:

**Comment by Conni2461** [comment]
Tami wanted me to comment here 🤣

**Pull request merged**